connectors/src/connectors/confluence/temporal/activities.ts

```diff
@@ -171,51 +181,44 @@ export async function confluenceListPageIdsInSpaceActivity({
 }
 
 async function upsertConfluencePageInDb(
-  connectionId: string,
-  dataSourceConfig: DataSourceConfig,
-  page: ConfluencePageWithBodyType
+  connectorId: ModelId,
+  page: ConfluencePageWithBodyType,
+  visitedAtMs: number
 ) {
-  const connector = await Connector.findOne({
-    where: {
-      connectionId,
-      dataSourceName: dataSourceConfig.dataSourceName,
-      workspaceId: dataSourceConfig.workspaceId,
-    },
-  });
-
-  if (!connector) {
-    throw new Error(`Connector not found (connectionId: ${connectionId})`);
-  }
-
   await ConfluencePage.upsert({
-    connectorId: connector.id,
+    connectorId,
```

Comment:
A bit balsy but OK assuming it fails if the connector got deleted by a data source deletion?

Reply:
I just moved the connector fetch before so the delay between fetching and doing this insertion is pretty low.